src/main/java/seedu/address/model/schedule/Event.java

```diff
@@ -250,29 +279,45 @@ public boolean willDateTimeCollideEvent(LocalDate date, LocalTime time) {
     }
 
     /**
-     * Returns an Event that happen at {@code date}.
+     * Splits the Event into multiple Events by date of occurrence and returns a list of the split Events
+     * that occur during the specified {@code date}
      *
      * @param date used to check.
-     * @return an Event for that particular date.
+     * @return a list of Events for that particular date.
      */
-    public Event getEventAtDate(LocalDate date) {
+    public List<Event> getEventsAtDate(LocalDate date) {
+        ArrayList<Event> eventsAtDate = new ArrayList<>();
         Event nextEvent = getNextRecurringEvent(date);
-        if (nextEvent.willDateCollide(date)) {
-            if (nextEvent.getDate().isBefore(date) && nextEvent.getEndDate().isAfter(date)) {
-                return new Event(nextEvent.eventDescription, date, LocalTime.MIDNIGHT,
-                        Duration.ofHours(24), recurFrequency);
-            } else if (nextEvent.getDate().isBefore(date) && nextEvent.getEndDate().isEqual(date)) {
-                Duration duration = Duration.between(LocalTime.MIDNIGHT, nextEvent.getEndTime());
-                return new Event(nextEvent.eventDescription, date, LocalTime.MIDNIGHT, duration, recurFrequency);
-            } else if (nextEvent.getDate().isEqual(date) && nextEvent.getEndDate().isAfter(date)) {
-                Duration duration = Duration.between(nextEvent.getTime(), LocalTime.MIDNIGHT).plusDays(1);
-                return new Event(nextEvent.eventDescription, date, nextEvent.getTime(), duration, recurFrequency);
-            } else {
-                return nextEvent;
+        if (date.isBefore(getDate()) || !willDateCollide(date)) {
```

Comment:
Perhaps you can add comments before the if statements to make the code more readable